game/lib/profile/completed_challenges_page.dart

```diff
@@ -59,12 +59,11 @@ class CompletedChallengesPage extends StatelessWidget {
             if (tracker == null || event == null) {
               continue;
             }
-            if (tracker.prevChallengeDates!.length !=
-                event.challenges!.length) {
+            if (tracker.prevChallenges!.length != event.challenges!.length) {
               continue;
             }
 
-            var completedDate = tracker.prevChallengeDates!.last;
+            var completedDate = tracker.prevChallenges.last.dateCompleted;
```

Comment:
Also asserting non-null and the last element

Reply:
Todo: catch error if last == null